pkg/monitors/haproxy/monitor.go

```diff
@@ -60,35 +59,22 @@ func (m *Monitor) Configure(conf *Config) (err error) {
 	}
 	interval := time.Duration(conf.IntervalSeconds) * time.Second
 	utils.RunOnInterval(m.ctx, func() {
-		ctx, cancel := context.WithTimeout(m.ctx, interval)
-		defer cancel()
 		var wg sync.WaitGroup
-		chs := make([]chan []*datapoint.Datapoint, 0)
+
 		for _, fn := range fetchFuncs {
 			fn := fn
-			ch := make(chan []*datapoint.Datapoint, 1)
 			wg.Add(1)
 			go func() {
-				defer close(ch)
-				defer wg.Done()
-				select {
-				case <-ctx.Done():
-					logger.Error(ctx.Err())
-					return
-				case ch <- fn(conf, pxs):
```

Comment:
I removed this because I don't think the context cancelling actually worked. My understanding is that the call to `fn` would always block until finished since it doesn't take in `ctx`. Having the select with `ch <- fn(conf, pxs)` I believe is for doing non-blocking send calls and `ch` would never block since it had size `1`.

Anyway it didn't seem necessary overall so I simplified things a bit.

Reply:
@bjsignalfx please confirm.

Reply:
@jaysfx No, I think the select statement blocks on both cases until there's stuff to read.  So, assuming fn() is long running and context cancellation is issued when timeout is reached, then the `case <-ctx.Done()` case will execute and the goroutine will return.

Reply:
I don't think that's the case. Here's a test case I came up with, see if you think it's correctly modeling it:

https://gist.github.com/jaysfx/284e87b34bc8fb8e0f7228a71c8859e8

You get the output 

```
before sleep
after sleep
slow returned
```

I don't think the select statement runs until _after_ the `slow` function returns. I think to have the intended effect ctx needs to be passed into the slow function so that it can be cancelled.

Reply:
To make it work I think you would have to call `fn` inside another separate goroutine and then do the select on a channel.

Reply:
To make it work I think you would have to call fn inside another separate goroutine and then do the select on a channel.

Reply:
Yes @jaysfx, you are right. If you can, please make this change instead.